### PR TITLE
Create protobuf generation GitHub action

### LINKED ISSUE #793: Consider generating Go types for OTLP in this repo
This will allow us to delete Go code from the `opentelemetry-proto` and make that repo cleaner in general.

See https://github.com/open-telemetry/opentelemetry-proto/issues/79#issuecomment-637881895 and the rest of the thread there.

### PR DESCRIPTION
**Pull request by evantorrie**
This is in preparation for automatically generating updated protobuf-derived `*.pb.go` files from a specific version of `https://github.com/open-telemetry/opentelemetry-proto` checked in as a git submodule in `internal/opentelemetry-proto` (coming in a subsequent PR). This allows source files in this repo to avoid having any direct dependency on `*.pb.go` files from `github.com/open-telemetry/opentelemetry-proto/go/gen` and thus should resolve #793.  Any sources within this repo which depend on the location of the protobuf generated files will need to be updated to reflect their new location which will be at `go.opentelemetry.io/otel/internal/opentelemetry-proto-gen/`.

### Synopsis of Github Action

Whenever a PR with an update to `internal/opentelemetry-proto` is created, this github action will be invoked on the PR and will

   * checkout the source from the PR, including the source from the submodule pointing to a specific commit of `https://github.com/open-telemetry/opentelemetry-proto`
   * run `make -f Makefile.proto protobuf`, which
      * rewrites the `option go_package = ...` lines in the `.proto` files to a package which resides at `go.opentelemetry.io/otel/internal/opentelemetry-proto-gen`, i.e. inside this repo.
      * uses a pinned version of the docker image `namely/protoc-all` to generate the new and changed `*.pb.go` files, copying them into the `internal/opentelemetry-proto-gen` directory.
   * use the `stefanzweifel/git-auto-commit-action@v4` github action to commit any changes to these files as a new commit in the PR.

The existing build action in CircleCI will run unchanged, but unlike this github action, will *not* populate the git submodule. Similarly, developers should run `make precommit` at the top-level of the repo with the git submodule at `internal/opentelemetry-proto` left *unpopulated* in all cases except when testing out an update to the .proto files. This avoids `go test ./...` pulling in dependencies needed in the `opentelemetry-proto` repo, but not needed for this repo.

### CODE REVIEW

**Comment by lizthegrey** [comment]
Love this change.

**Comment by evantorrie** [comment]
For an example of what happens after a PR with a new version of the git submodule, you can take a look at my tests here: https://github.com/evantorrie/opentelemetry-go/pull/35
The github action commit is this one:  https://github.com/evantorrie/opentelemetry-go/pull/35/commits/115349fb9b10885df437fc903abc12b3c46012cd

The PR itself still shows as failed because I don't have anything larger than the default "medium" CircleCI runner size for my personal account, so the links/tests of the normal build fail, but this should not be a problem on this account.

File: Makefile.proto [hunk 1]
```diff
@@ -0,0 +1,71 @@
+# Copyright The OpenTelemetry Authors  -*- mode: makefile; -*-
```
Comment (MrAlias) [inline Makefile.proto:1]: Is this intentional?

```suggestion
# Copyright The OpenTelemetry Authors
```

Or maybe?

```suggestion
#   -*- mode: makefile; -*-
# Copyright The OpenTelemetry Authors
```

Comment (evantorrie) [reply Makefile.proto:1]: It is intentional - emacs recognizes `Makefile` or `makefile` as filenames to automatically enter Makefile mode, but not something that ends in `.proto`. The file-local variables can be anywhere on the first line.. so I guess it should probably not be confused with the Copyright notice.

But I'm also open to just removing it and remembering to go into Makefile mode when I open Makefile.proto.

Comment (MrAlias) [reply Makefile.proto]: Ah, gotcha. I think the accepted change sounds appropriate than.

**Pull request merged**